backend/module/bot/slackbot/slackbot_helper.go

```diff
@@ -0,0 +1,51 @@
+package slackbot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/slack-go/slack"
+)
+
+// the default bot reply if the command is not matched
+const (
+	HelpIntro = "Hi! I respond to the following commands:\n"
+	// TODO: (sperry) this should be auto generated based on what we allow users to do via the bot
+	HelpDetails = "```help: Print this message\nhello: I will greet you with Hello, World!```"
+)
+
+var (
+	// example: <@UVWXYZ123>
+	slackUserIdRegex = regexp.MustCompile(`^<@(\w+)>`)
+	spaceRegex       = regexp.MustCompile(`\s+`)
+)
+
+// TrimUserId removes the user id of the bot from the command string
+// Commands issued in a channel (instead of in a DM) will start with the user ID of the bot. Example: '<@UVWXYZ123> describe pod'
+// and should be removed before matching the command
+func TrimUserId(text string) string {
+	trimmed := slackUserIdRegex.ReplaceAllString(text, "bot_user_id")
+	// removes an extra whitespace so that there isn't a leading whitespace in the beginning of the new string
+	return strings.Replace(trimmed, "bot_user_id ", "", 1)
+}
+
+// TrimRedundantSpaces replaces trailing/leading and duplicate whitespace
+// this is a sanitization check before matching the command
+func TrimRedundantSpaces(text string) string {
+	text = spaceRegex.ReplaceAllString(text, " ")
+	return strings.TrimSpace(text)
+}
+
+// DefaultHelp returns the default help message is the command isn't matched
+func DefaultHelp() string {
+	return fmt.Sprintf("%s\n%s", HelpIntro, HelpDetails)
+}
```

Comment:
this should be in the service, i get it's a proof of concept though, so maybe it's fine for now.

Reply:
i put it at the module b/c message formatting could look different among providers (ie Slack vs Discord) and I wanted to the the slack code block formatting for the help message

Reply:
i thought about it for a bit and i don't think the module would be involved in formatting, i think that should still happen in the service. the module just handles ingest and response (either async or sync depending on which service method it calls), and the service handles everything else.

Reply:
got it, shuffled things around. the current flow is that the service will have some provider specific entry points that then all route to the generic command that matches a message string to a clutch api call.

Reply:
convo moved to https://github.com/lyft/clutch/pull/1282#pullrequestreview-633711821